testQueryRandomGeoCollection fails

I can't reproduce this locally after several runs.

```
./gradlew :server:test --tests "org.elasticsearch.search.geo.GeoShapeQueryTests.testQueryRandomGeoCollection" \
-Dtests.seed=62A08EC8DF6B4952 \
-Dtests.security.manager=true \
-Dtests.locale=en-US \
-Dtests.timezone=Europe/Sofia \
-Dcompiler.java=12 \
-Druntime.java=8
```

```
org.elasticsearch.search.geo.GeoShapeQueryTests > testQueryRandomGeoCollection FAILED
    java.lang.AssertionError: Count is 0 hits but 1 was expected.  Total shards: 1 Successful shards: 1 & 0 shard failures:
        at __randomizedtesting.SeedInfo.seed([62A08EC8DF6B4952:7A77A51382991671]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount(ElasticsearchAssertions.java:256)
        at org.elasticsearch.search.geo.GeoShapeQueryTests.testQueryRandomGeoCollection(GeoShapeQueryTests.java:401)
```

CI: https://scans.gradle.com/s/kspyg36gvpjlw

--- a/server/src/main/java/org/elasticsearch/index/query/VectorGeoShapeQueryProcessor.java
+++ b/server/src/main/java/org/elasticsearch/index/query/VectorGeoShapeQueryProcessor.java
@@ -66,7 +66,7 @@ public class VectorGeoShapeQueryProcessor implements AbstractGeometryFieldMapper
         if (processedShape == null) {
             return new MatchNoDocsQuery();
         }
-        return queryShape.visit(new ShapeVisitor(context, fieldName, relation));
+        return processedShape.visit(new ShapeVisitor(context, fieldName, relation));
     }
 
     private class ShapeVisitor implements GeometryVisitor<Query, RuntimeException> {

